prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -596,8 +597,17 @@ func upstreamImageVersionResolver(
 		imageFullPath := imageHost + "/" + imageName + ":" + currentTag
 		for prefix, version := range upstreamVersions {
 			if strings.HasPrefix(imageFullPath, prefix) {
-				imageBumperCli.AddToCache(imageFullPath, version)
-				return version, nil
+				exists, err := imageBumperCli.TagExists(imageHost, imageName, version)
```

Comment:
Would it make more sense to do this at a higher level than the upstream image version resolver so that this logic applies to all resolvers? We should never bump to an image that doesn't exist.

Reply:
The three possibilities for TargetVersion are "upstream" "latest" and "other". This handles "upstream". "latest" already handles this issue implicitly. Anything other than upstream or latest is a hard override which will just set the image to whatever the TargetVersion is, even if it is not a valid tag. None of the previous autobumpers performed any checks for this option. I assumed we wanted users to be able to use this hard override target version to change the image tag to whatever they wished without restriction. If this was just an oversight, I can add the check there as well?

Reply:
The only time I can imagine wanting to bump to a tag that doesn't exist is if the image was private and the tag did actually exist, but the autobumper didn't have permission to list tags. The right way to solve that would probably be to authorize the autobumper, but I guess it doesn't matter much since we don't use the "other" mode anyways. There might be other reasons for a hard override I'm not thinking of too.
/shrug

Reply:
I mostly use the hard override mode for testing. I think it might be best to go back in another PR and formalize this by adding a "enforce image exists" option. For now I think this is fine though.